solidity/contracts/bridge/VendingMachine.sol

```diff
@@ -0,0 +1,250 @@
+        unmintFeeChangeInitiated = 0;
+    }
+
+    /// @notice Allows the Governance to begin vending machine update process.
+    ///         The update process needs to be finalized with a call to
+    ///         `finalizeVendingMachineUpdate` function after the
+    ///         `GOVERNANCE_DELAY` passes.
+    /// @param _newVendingMachine The new vending machine address
+    function beginVendingMachineUpdate(address _newVendingMachine)
+        external
+        onlyOwner
+    {
+        require(
+            _newVendingMachine != address(0),
+            "New VendingMachine can not be zero address"
+        );
+
+        /* solhint-disable-next-line not-rely-on-time */
+        emit VendingMachineUpdateStarted(_newVendingMachine, block.timestamp);
+        newVendingMachine = _newVendingMachine;
+        /* solhint-disable-next-line not-rely-on-time */
+        vendingMachineUpdateInitiated = block.timestamp;
+    }
+
+    /// @notice Allows the Governance to finalize vending machine update process.
+    ///         The update process needs to be first initiated with a call to
+    ///         `beginVendingMachineUpdate` and the `GOVERNANCE_DELAY` needs to
+    ///         pass. Once the update is finalized, the new vending machine will
+    ///         become an owner of TBTC v2 token.
+    function finalizeVendingMachineUpdate()
+        external
+        onlyOwner
+        onlyAfterGovernanceDelay(vendingMachineUpdateInitiated)
+    {
+        emit VendingMachineUpdated(newVendingMachine);
+        //slither-disable-next-line reentrancy-no-eth
+        tbtcV2.transferOwnership(newVendingMachine);
+        newVendingMachine = address(0);
+        vendingMachineUpdateInitiated = 0;
+    }
+
+    /// @notice Get the remaining time that needs to pass until unmint fee
+    ///         update can be finalized by the Governance. If the update has
+    ///         not been initiated, the function reverts.
+    function getRemainingUnmintFeeUpdateTime() external view returns (uint256) {
+        return
+            GovernanceUtils.getRemainingChangeTime(
+                unmintFeeChangeInitiated,
+                GOVERNANCE_DELAY
+            );
+    }
+
+    /// @notice Get the remaining time that needs to pass until vending machine
+    ///         update can be finalized by the Governance. If the update has not
+    ///         been initiated, the function reverts.
+    function getRemainingVendingMachineUpdateTime()
+        external
+        view
+        returns (uint256)
+    {
+        return
+            GovernanceUtils.getRemainingChangeTime(
+                vendingMachineUpdateInitiated,
+                GOVERNANCE_DELAY
+            );
+    }
+
+    /// @notice Returns the fee that needs to be paid to the `VendingMachine` to
+    ///         unmint the given amount of TBTC v2 back into TBTC v1.
+    function unmintFeeFor(uint256 amount) public view returns (uint256) {
+        return (amount * unmintFee) / FLOATING_POINT_DIVISOR;
```

Comment:
Don't we need to use SafeMath for calculations?

Reply:
Not with Solidity 8 [checked arithmetics](https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/).